PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterCharacterScripts/Sound.rbxmx

```diff
@@ -0,0 +1,381 @@
+		local hVel = Head.Velocity + Vector3.new(0,-Head.Velocity.Y,0)
+		return hVel.magnitude	
+	end;
+	
+	--Gets the vertical (y) velocity magnitude of the given part
+	VerticalVelocityMagnitude = function(Head)
+		return math.abs(Head.Velocity.Y)
+	end;
+	
+	--Checks if Sound.Playing is usable
+	SoundIsV2 = function(sound)
+		return sound.IsLoaded
+	end;
+	
+	--Setting Playing/TimePosition values directly result in less network traffic than Play/Pause/Resume/Stop
+	--If these properties are enabled, use them.
+	Play = function(sound)		
+		if Util.SoundIsV2(sound) then
+			if sound.TimePosition ~= 0 then
+				sound.TimePosition = 0
+			end
+			if not sound.IsPlaying then
+				sound.Playing = true
+			end
+		else
+			sound:Play()
+		end
+		
+	end;
+	Pause = function(sound)
+		if Util.SoundIsV2(sound) then
+			if sound.IsPlaying then
+				sound.Playing = false
+			end
+		else
+			sound:Pause()
+		end
+	end;
+	Resume = function(sound)
+		if Util.SoundIsV2(sound) then
+			if not sound.IsPlaying then
+				sound.Playing = true
+			end
+		else
+			sound:Resume()
+		end
+	end;
+	Stop = function(sound)
+		if Util.SoundIsV2(sound) then
+			if sound.IsPlaying then
+				sound.Playing = false
+			end
+			if sound.TimePosition ~= 0 then
+				sound.TimePosition = 0
+			end
+		else
+			sound:Stop()
+		end
+	end;
+}
+
+do
+	-- List of all active Looped sounds
+	local activeLoopedSounds = {}
+	
+	-- Last seen Enum.HumanoidStateType
+	local activeState = nil
+	
+	-- Verify and set that the sound's .Looped property is the value of "looped"
+	function verifyAndSetLoopedForSound(sound, looped)
+		if sound.Looped ~= looped then
+			sound.Looped = looped
+		end
+	end
+	
+	-- Verify and set that "sound" is in "activeLoopedSounds".
+	function setSoundInActiveLooped(sound)
+		for i=1, #activeLoopedSounds do
+			if activeLoopedSounds[i] == sound then
+				return
+			end
+		end	
+		table.insert(activeLoopedSounds,sound)
+	end
+	
+	-- Stop all active looped sounds except parameter "except". If "except" is not passed, all looped sounds will be stopped.
+	function stopActiveLoopedSoundsExcept(except)
+		for i=#activeLoopedSounds,1,-1 do
+			if activeLoopedSounds[i] ~= except then
+				Util.Pause(activeLoopedSounds[i])			
+				table.remove(activeLoopedSounds,i)	
+			end
+		end
+	end
+	
+	-- Table of Enum.HumanoidStateType to handling function
+	local stateUpdateHandler = {
+		[Enum.HumanoidStateType.Dead] = function()
+			stopActiveLoopedSoundsExcept()
+			local sound = Sounds[SFX.Died]
+			verifyAndSetLoopedForSound(sound,false)
+			Util.Play(sound)
+		end;
+		
+		[Enum.HumanoidStateType.RunningNoPhysics] = function()
+			stateUpdated(Enum.HumanoidStateType.Running)
+		end;
+		
+		[Enum.HumanoidStateType.Running] = function()	
+			local sound = Sounds[SFX.Running]
+			verifyAndSetLoopedForSound(sound,true)
+			stopActiveLoopedSoundsExcept(sound)
+			
+			if Util.HorizontalVelocityMagnitude(Head) > 0.5 then
+				if not sound.IsPlaying then
```

Comment:
is this if check really necessary? I don't think it will save much time, and if it does why not just put it in util?